src/hotspot/share/jeandle/jeandleCompiledCode.cpp

```diff
@@ -406,11 +402,126 @@ static VMReg resolve_vmreg(const StackMapParser::LocationAccessor& location, Sta
+  case T_INT: {
+    if (is_constant) {
+      array->at_put_grow(encode._index, new ConstantIntValue(StackMapUtil::getConstantUint(stackmaps, location)));
+    } else {
+      array->at_put_grow(encode._index,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::normal, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::normal, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_LONG: {
+    // TODO: we need 2 slots for long value
+    if (is_constant) {
+      array->at_put_grow(encode._index, new ConstantLongValue(StackMapUtil::getConstantUlong(stackmaps, location)));
+    } else {
+      array->at_put_grow(encode._index,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::lng, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::lng, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_FLOAT: {
+    if (is_constant) {
+      array->at_put_grow(encode._index, new ConstantIntValue(jint_cast(StackMapUtil::getConstantFloat(stackmaps, location))));
+    } else {
+      array->at_put_grow(encode._index,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::normal, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::normal, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_DOUBLE: {
+    // TODO: we need 2 slots for double value
+    if (is_constant) {
+      array->at_put_grow(encode._index, new ConstantDoubleValue(StackMapUtil::getConstantDouble(stackmaps, location)));
+    } else {
+      array->at_put_grow(encode._index,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::dbl, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::dbl, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_OBJECT: {
+    if (is_constant) {
+      // array->at_put_grow(encode._index, new ConstantOopReadValue(StackMapUtil::getConstantDouble(stackmaps, location)));
+    } else {
+      array->at_put_grow(encode._index,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::oop, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::oop, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_ILLEGAL: {
+    uint32_t val = StackMapUtil::getConstantUint(stackmaps, location);
+    assert(val == 0, "must be zero for T_VOID");
+    // put an illegal value
+    array->at_put_grow(encode._index, new LocationValue(Location()));
+    break;
+  }
```

Comment:
As I understand, long and double values are represented by a long/double value and an illegal value in JeandleVMState. Those illegal values are also encoded to deopt bundle and then decoded here. So maybe we don't need to add an extra empty slot for long and double values. But refer to c2's implementation, the empty slot should go before the real long/double values in ScopeValue array.